File: src/main/java/ffblockly/meta/FFBlock.java
package ffblockly.meta;


import java.util.List;

/**
 * Created by Jnorthrup on 9/13/2016.
 */
public interface FFBlock {
    Boolean isInputsInline();

    ffblockly.meta.FilterListing.FilterSignature getPreviousStatement();

    ffblockly.meta.FilterListing.FilterSignature getNextStatement();

    int getColour();

    String getTooltip();


    String getHelpUrl();

    String getType();

    String getMessage0();

    List<ffblockly.meta.FFBlocklyFieldValue> getArgs0();

}

File: src/main/java/ffblockly/meta/FFBlock.kt
package ffblockly.meta

/**
 * Created by Jnorthrup on 9/13/2016.
 */
interface FFBlock {
    val isInputsInline: Boolean?
    val previousStatement: FilterListing.FilterSignature?
    val nextStatement: FilterListing.FilterSignature?
    val colour: Int
    val tooltip: String?
    val helpUrl: String?
    val type: String?
    val message0: String?
    val args0: List<FFBlocklyFieldValue?>?
}